MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup.

swnsma opened the pull request:
### Description (*)
https://github.com/magento/inventory/issues/2826

### Fixed Issues (if relevant)
1. magento/inventory#2826: Hide store pickup during order submissions if one of the products is not available for pickup

Commit:
MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup.
Create API declaration.

Commit:
MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup.
Add draft implementation.

Commit:
MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup.
Add implementation for search intersection of pickup locations for requested product info.
Change interfaces implementation - add ProductInfoInterface.

novikor commented on InventoryInStorePickupShipping/Model/ResourceModel/GetPickupLocationIntersectionForSkues.php:
```diff
@@ -0,0 +1,84 @@
+    /**
+     * @var SourceItem
+     */
+    private $sourceItemResource;
```
Please add whitespaces between fields.

swnsma replied:
Done.

novikor commented on InventoryInStorePickupShipping/Model/ResourceModel/GetPickupLocationIntersectionForSkues.php:
```diff
@@ -0,0 +1,84 @@
+/**
+ * Provides list of Source Codes which have all requested products assigned.
+ */
+class GetPickupLocationIntersectionForSkues
```
Did you mean `GetPickupLocationIntersectionForSkus`?

novikor replied:
I am not sure if this class is related to any intersections for real.
It searches active pickup locations by SKUs, isn't it?

swnsma replied:
It provides list of pickup locations which have intersection of assignment of provided in request SKUs.

novikor replied:
What about `GetAsignedPickupLocationsBySkus` or something like that? The class name should not sound like it contains something from the quantum physics inside

swnsma replied:
I think name `GetAsignedPickupLocationsBySkus` does not represents what it actually do.

novikor replied:
Please use `GetPickupLocationIntersectionForSkus`

swnsma replied:
Done!

novikor commented on InventoryInStorePickupShipping/Model/GetAvailableLocationsForPickup.php:
```diff
@@ -0,0 +1,106 @@
+     */
+    public function execute(RequestInterface $request): array
+    {
+        $skues = [];
```
may be `$skus`?

swnsma replied:
Done!

Commit:
MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup.
Apply code review changes.

swnsma committed:
Merge branch '1.2-develop' into MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup

Commit:
MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup.
Add api-functional tests coverage.

Commit:
Merge branch 'MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup' of github.com:magento/inventory into MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup

novikor committed:
Merge branch '1.2-develop' into MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup

Commit:
MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup.
Add api mark for interfaces.

novikor committed:
Merge branch '1.2-develop' into MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup

Commit:
Merge branch 'MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup' of github.com:magento/inventory into MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup

Commit:
magento/inventory#2839: Used extension attributes

Commit:
magento/inventory#2839: Added frontend part

Commit:
magento/inventory#2839: Moved to filters extensions attributes

Commit:
magento/inventory#2839: Moved plugin to filters resolver

Commit:
MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup.
Remove plugin - move logic to the chain.
Migrate tests to the new module.

Commit:
Merge branch 'MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup' of github.com:magento/inventory into MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup

Commit:
MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup.
Add missed param. Move extension attributes to API module.

Commit:
MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup.
Remove extension attributes from filter group.
Add extension attributes to search request.

Commit:
MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup.
Revert pathes in tests.

swnsma committed:
Merge branch '1.2-develop' into MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup

Commit:
MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup.
Fix static tests.

Commit:
Merge branch 'MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup' of github.com:magento/inventory into MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup

sidolov commented on InventoryInStorePickup/Model/ResourceModel/GetPickupLocationIntersectionForSkus.php:
```diff
@@ -0,0 +1,87 @@
+     * @return array
+     * @throws LocalizedException
+     */
+    public function execute(array $skus)
```
Please, add strict type

swnsma replied:
Done.

sidolov commented on InventoryInStorePickup/Model/SearchCriteria/ResolveIntersection.php:
```diff
@@ -0,0 +1,63 @@
+        }
+
+        $codes = $this->getPickupLocationIntersectionForSkus->execute($skus);
+        $codes = implode(',', $codes);
```
SearchCriteria should correctly work with arrays

swnsma replied:
Should we remove strict types or should allow only arrays to be passed?
In case of last we need implement something like 'mapping' to allow arrays be passed (like 'in' or 'nin' conditions) and values which need to be interpreted into string (like 'eq', 'gt' etc).

Commit:
MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup.
Add missed return type.

sidolov commented:
@magento run Functional Tests CE,Functional Tests EE,Functional Tests B2B

Commit:
magento/inventory#2839: Minor fixes

Commit:
Merge remote-tracking branch 'origin/1.2-develop' into MSI-2826-Hide-store-pickup-during-order-submissions-if-one-of-the-products-is-not-available-for-pickup
